File: campa-distributor-system/mobile-app/lib/models/product.dart
class Product {
  final int id;
  final String name;
  final String sku;
  final double price;
  final int stockQuantity;
  final int bottlesPerCrate;

  Product({
    required this.id,
    required this.name,
    required this.sku,
    required this.price,
    required this.stockQuantity,
    this.bottlesPerCrate = 24, // Default
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      name: json['name'],
      sku: json['sku'],
      price: double.tryParse(json['price'].toString()) ?? 0.0,
      stockQuantity: json['stockQuantity'] ?? 0,
      bottlesPerCrate: json['bottlesPerCrate'] ?? 24,
    );
  }
}

File: campa-distributor-system/backend/models/Product.js
module.exports = (sequelize, DataTypes) => {
    const Product = sequelize.define('Product', {
        id: {
            type: DataTypes.INTEGER,
            autoIncrement: true,
            primaryKey: true,
        },
        name: {
            type: DataTypes.STRING,
            allowNull: false,
        },
        sku: {
            type: DataTypes.STRING,
            unique: true,
        },
        price: {
            type: DataTypes.DECIMAL(10, 2),
            allowNull: false,
        },
        gstPercentage: {
            type: DataTypes.DECIMAL(5, 2),
            defaultValue: 18.00,
        },
        hsnCode: {
            type: DataTypes.STRING,
            comment: 'HSN Code for GST',
        },
        stockQuantity: {
            type: DataTypes.INTEGER,
            defaultValue: 0,
        },
        bottlesPerCrate: {
            type: DataTypes.INTEGER,
            defaultValue: 24,
        },
        groupName: {
            type: DataTypes.STRING,
            comment: 'Product Group e.g., ORANGE, LEMON',
        },
        isActive: {
            type: DataTypes.BOOLEAN,
            defaultValue: true,
        },
    });

    return Product;
};
